src/store/index.js

```diff
@@ -0,0 +1,29 @@
+import Vue from "vue";
+import Vuex from "vuex";
+import Http from "../api/apiInterface.js";
+
+Vue.use(Vuex);
+
+const store = new Vuex.Store({
+  state: {
+    productList: [],
+  },
+  mutations: {
+    setProductList(state, payload) {
+      state.productList = payload;
+    },
+  },
+  actions: {
+    async fetchProductList(context) {
+      const data = await Http.get("/products");
+      context.commit("setProductList", await data);
```

Comment:
Is the `await` for data needed here? 🤔 It does not look like it should be a promise. 🤔

Reply:
Needed. From ` Http.get` I am receiving promise.